contracts/v2/marketplace/TokenMarketplaceV2.sol

```diff
@@ -0,0 +1,496 @@
+    address indexed _buyer,
+    address indexed _seller,
+    uint256 _price
+  );
+
+  event BidPlaced(
+    uint256 indexed _tokenId,
+    address indexed _currentOwner,
+    address indexed _bidder,
+    uint256 _amount
+  );
+
+  event BidWithdrawn(
+    uint256 indexed _tokenId,
+    address indexed _bidder
+  );
+
+  event BidAccepted(
+    uint256 indexed _tokenId,
+    address indexed _currentOwner,
+    address indexed _bidder,
+    uint256 _amount
+  );
+
+  event BidRejected(
+    uint256 indexed _tokenId,
+    address indexed _currentOwner,
+    address indexed _bidder,
+    uint256 _amount
+  );
+
+  event AuctionEnabled(
+    uint256 indexed _tokenId,
+    address indexed _auctioneer
+  );
+
+  event AuctionDisabled(
+    uint256 indexed _tokenId,
+    address indexed _auctioneer
+  );
+
+  struct Offer {
+    address bidder;
+    uint256 offer;
+  }
+
+  struct Listing {
+    uint256 price;
+    address seller;
+  }
+
+  // Min increase in bid/list amount
+  uint256 public minBidAmount = 0.04 ether;
+
+  // Interface into the KODA world
+  IKODAV2Methods public kodaAddress;
+
+  // KO account which can receive commission
+  address public koCommissionAccount;
+
+  uint256 public artistRoyaltyPercentage = 100;
+  uint256 public platformFeePercentage = 25;
+
+  // Token ID to Offer mapping
+  mapping(uint256 => Offer) public offers;
+
+  // Token ID to Listing
+  mapping(uint256 => Listing) public listings;
+
+  // Explicitly disable sales for specific tokens
+  mapping(uint256 => bool) public disabledTokens;
+
+  ///////////////
+  // Modifiers //
+  ///////////////
+
+  modifier onlyWhenOfferOwner(uint256 _tokenId) {
+    require(offers[_tokenId].bidder == msg.sender, "Not offer maker");
+    _;
+  }
+
+  modifier onlyWhenTokenExists(uint256 _tokenId) {
+    require(kodaAddress.exists(_tokenId), "Token does not exist");
+    _;
+  }
+
+  modifier onlyWhenBidOverMinAmount(uint256 _tokenId) {
+    require(msg.value >= offers[_tokenId].offer.add(minBidAmount), "Offer not enough");
+    _;
+  }
+
+  modifier onlyWhenTokenAuctionEnabled(uint256 _tokenId) {
+    require(!disabledTokens[_tokenId], "Token not enabled for offers");
+    _;
+  }
+
+  /////////////////
+  // Constructor //
+  /////////////////
+
+  // Set the caller as the default KO account
+  constructor(IKODAV2Methods _kodaAddress, address _koCommissionAccount) public {
+    kodaAddress = _kodaAddress;
+    koCommissionAccount = _koCommissionAccount;
+    super.addAddressToWhitelist(msg.sender);
+  }
+
+  //////////////////////////
+  // User Bidding Actions //
+  //////////////////////////
+
+  function placeBid(uint256 _tokenId)
+  public
+  payable
+  whenNotPaused
+  nonReentrant
+  onlyWhenTokenExists(_tokenId)
+  onlyWhenBidOverMinAmount(_tokenId)
+  onlyWhenTokenAuctionEnabled(_tokenId)
+  {
+    require(!isContract(msg.sender), "Unable to place a bid as a contract");
+    _refundHighestBidder(_tokenId);
+
+    offers[_tokenId] = Offer(msg.sender, msg.value);
```

Comment:
use the `{}` way of setting up like Listing. Minor.

Reply:
Added